client/src/components/Schedule/CalendarView/utils/DataFuncs.ts

```diff
@@ -0,0 +1,16 @@
+import { filterByDate } from './DateFuncs';
+import { CourseEvent } from 'services/course';
+import { EventTypeColor} from 'components/Schedule/model';
+import {Moment} from 'moment';
+
+
+export function getListData(calendarCellDate: Moment, data: CourseEvent[], timeZone: string) {
+  const listData: { color: string; name: string; key: number }[] = [];
+
+  filterByDate(calendarCellDate, data, timeZone).forEach((el: CourseEvent) => {
+    const tagColor = EventTypeColor[el.event.type as keyof typeof EventTypeColor];
```

Comment:
I'm sure, that the construction `[el.event.type as keyof typeof EventTypeColor]` can be rewritten to something far simpler.

Reply:
I didn't succeed(